Tests/kaas/plugin/plugin_cluster_stacks.py

```diff
@@ -0,0 +1,366 @@
+import os
+import yaml
+import shutil
+import subprocess
+import base64
+import time
+import logging
+from pytest_kind import KindCluster
+from interface import KubernetesClusterPlugin
+
+logger = logging.getLogger("PluginClusterStacks")
+
+
+# Helper functions
+def wait_for_capi_pods_ready(timeout=240, interval=15):
+    """
+    Waits for all CAPI pods in specific namespaces to reach the 'Running' state with all containers ready.
+
+    :param timeout: Total time to wait in seconds before giving up.
+    :param interval: Time to wait between checks in seconds.
+    """
+    namespaces = [
+        "capi-kubeadm-bootstrap-system",
+        "capi-kubeadm-control-plane-system",
+        "capi-system",
+    ]
+
+    start_time = time.time()
+
+    while time.time() - start_time < timeout:
+        all_pods_ready = True
+
+        for namespace in namespaces:
+            try:
+                # Get pod status in the namespace
+                result = subprocess.run(
+                    f"kubectl get pods -n {namespace} -o=jsonpath='{{range .items[*]}}{{.metadata.name}} {{.status.phase}} {{range .status.containerStatuses[*]}}{{.ready}} {{end}}{{\"\\n\"}}{{end}}'",
+                    shell=True, capture_output=True, text=True, check=True
+                )
+
+                if result.returncode == 0:
+                    pods_status = result.stdout.strip().splitlines()
+                    for pod_status in pods_status:
+                        pod_info = pod_status.split()
+                        pod_name, phase, *readiness_states = pod_info
+
+                        # Check pod phase and all containers readiness
+                        if phase != "Running" or "false" in readiness_states:
+                            all_pods_ready = False
+                            logger.info(f"Pod {pod_name} in {namespace} is not ready. Phase: {phase}, Ready: {readiness_states}")
+                else:
+                    logger.info(f"Error fetching pods in {namespace}: {result.stderr}")
+                    all_pods_ready = False
+
+            except subprocess.CalledProcessError as error:
+                logger.error(f"Error checking pods in {namespace}: {error}")
+                all_pods_ready = False
+
+        if all_pods_ready:
+            logger.info("All CAPI system pods are ready.")
+            return True
+
+        logger.info("Waiting for all CAPI pods to become ready...")
+        time.sleep(interval)
+
+    raise TimeoutError(f"Timed out after {timeout} seconds waiting for CAPI and CAPO system pods to become ready.")
+
+
+def wait_for_cso_pods_ready(timeout=240, interval=15):
```

Comment:
This seems like a variation of the function above (if not literally copy and paste), only with different `namespaces`. The function above should take `namespaces` as parameter, renamed `wait_for_pods`, and then `wait_for_cso_pods_ready` and `wait_for_capi_pods_ready` be implemented as a simple call to `wait_for_pods`.

Reply:
I will try to merge these two functions into the one `wait_for_pods`